src/NuGet.Clients/NuGet.PackageManagement.UI/Options/PackageSourceCheckedListBox.cs

```diff
@@ -6,14 +6,46 @@
 using System.Runtime.InteropServices;
 using System.Windows.Forms;
 using System.Windows.Forms.VisualStyles;
+using Microsoft.VisualStudio.Imaging;
+using Microsoft.VisualStudio.Imaging.Interop;
+using Microsoft.VisualStudio.Shell.Interop;
 using NuGet.PackageManagement.UI;
 using NuGet.VisualStudio.Internal.Contracts;
+using GelUtilities = Microsoft.Internal.VisualStudio.PlatformUI.Utilities;
+using NuGet.Configuration;
 
 namespace NuGet.Options
 {
     internal class PackageSourceCheckedListBox : CheckedListBox
     {
         public Size CheckBoxSize { get; set; }
+        private static IVsImageService2 ImageService
+        {
+            get
+            {
+                return (IVsImageService2)Microsoft.VisualStudio.Shell.Package.GetGlobalService(typeof(SVsImageService));
+            }
+        }
+
+        public static Icon WarningIcon
```

Comment:
Properties are just methods with some syntactic sugar, and static methods are just methods that don't have a `this` "variable". In order to use a `static` for caching, it must be a static field.

Also, some people (myself included) believe that properties should not have "business logic", although I don't think this is in our team coding guidelines. It's much easier to understand and optimize for both correctness and performance when we can assume that fields and properties will return values already in memory (and don't change state), and methods are called when internal state is changed or when potentially expensive calculations need to be made.

So, I suggest removing these two static properties, add a `private static Icon _warningIcon;`, and a `private Icon GetWarningIcon()` that returns `_warningIcon` when it's not null, and gets the icon from the image service when it is null.